src/ReportPortal.Client/Abstractions/Resources/ILaunchResource.cs

```diff
@@ -24,55 +31,120 @@ public interface ILaunchResource
         /// <returns>A message from service.</returns>
         Task<LaunchFinishedResponse> FinishAsync(string uuid, FinishLaunchRequest request);
 
+        /// <summary>
+        /// Finishes specified launch.
+        /// </summary>
+        /// <param name="uuid">UUID of specified launch.</param>
+        /// <param name="request">Information about representation of launch to finish.</param>
+        /// <param name="cancellationToken">Cancellation token.</param>
+        /// <returns>A message from service.</returns>
+        Task<LaunchFinishedResponse> FinishAsync(string uuid, FinishLaunchRequest request, CancellationToken cancellationToken);
+
         /// <summary>
         /// Deletes specified launch.
         /// </summary>
         /// <param name="id">ID of the launch to delete.</param>
         /// <returns>A message from service.</returns>
         Task<MessageResponse> DeleteAsync(long id);
 
+        /// <summary>
+        /// Deletes specified launch.
+        /// </summary>
+        /// <param name="id">ID of the launch to delete.</param>
+        /// <param name="cancellationToken">Cancellation token.</param>
+        /// <returns>A message from service.</returns>
+        Task<MessageResponse> DeleteAsync(long id, CancellationToken cancellationToken);
+
         /// <summary>
         /// Returns specified launch by ID.
         /// </summary>
         /// <param name="id">ID of the launch to retrieve.</param>
         /// <returns>A representation of launch.</returns>
         Task<LaunchResponse> GetAsync(long id);
 
+        /// <summary>
+        /// Returns specified launch by ID.
+        /// </summary>
+        /// <param name="id">ID of the launch to retrieve.</param>
+        /// <param name="cancellationToken">Cancellation token.</param>
+        /// <returns>A representation of launch.</returns>
+        Task<LaunchResponse> GetAsync(long id, CancellationToken cancellationToken);
+
         /// <summary>
         /// Returns specified launch by UUID.
         /// </summary>
         /// <param name="uuid">UUID of the launch to retrieve.</param>
         /// <returns>A representation of launch.</returns>
         Task<LaunchResponse> GetAsync(string uuid);
 
+        /// <summary>
+        /// Returns specified launch by UUID.
+        /// </summary>
+        /// <param name="uuid">UUID of the launch to retrieve.</param>
+        /// <param name="cancellationToken">Cancellation token.</param>
+        /// <returns>A representation of launch.</returns>
+        Task<LaunchResponse> GetAsync(string uuid, CancellationToken cancellationToken);
+
         /// <summary>
         /// Returns a list of launches for current project.
         /// </summary>
         /// <param name="filterOption">Specified criterias for retrieving launches.</param>
         /// <returns>A list of launches.</returns>
         Task<Content<LaunchResponse>> GetAsync(FilterOption filterOption = null);
```

Comment:
I propose to make it also overloaded rather than using default parameter. It keeps consistency.